addons/web/static/src/js/core/owl_popover.js

```diff
@@ -0,0 +1,234 @@
+            // try all other positions and find one that does
+            const position = this.orderedPositions
+                .slice(positionIndex)
+                .concat(this.orderedPositions.slice(0, positionIndex))
+                .map((pos) => positioningData[pos])
+                .find((pos) => {
+                    this.popoverRef.el.style.top = `${pos.top}px`;
+                    this.popoverRef.el.style.left = `${pos.left}px`;
+                    return this._isInViewport(this.popoverRef.el);
+                });
+
+            // remove all positioning classes
+            this.orderedPositions.forEach((pos) => {
+                this.popoverRef.el.classList.remove(`o_popover--${pos}`);
+            });
+
+            /* apply the found position ('position' props or another position
+            that fits the viewport) if it fits the viewport, otherwise
+            fallback to 'bottom' if no position fits the viewport */
+            if (position) {
+                this.popoverRef.el.classList.add(`o_popover--${position.name}`);
+            } else {
+                this.popoverRef.el.style.top = `${positioningData.bottom.top}px`;
+                this.popoverRef.el.style.left = `${positioningData.bottom.left}px`;
+                this.popoverRef.el.classList.add(`o_popover--bottom`);
+            }
+        }
+
+        //--------------------------------------------------------------------------
+        // Private
+        //--------------------------------------------------------------------------
+
+        /**
+         * Send an event signaling that the popover must be closed.
+         * @private
+         */
+        _close() {
+            this.trigger('close-popover');
+        }
+
+        /**
+         * Check if the element is in the viewport.
+         * @param {HTMLElement}
+         * @returns {Boolean} True if the popover fits in the viewport, false otherwise.
+         */
+        _isInViewport(element) {
+            const rect = element.getBoundingClientRect();
+            const html = document.documentElement;
+            return (
+                rect.top >= 0 &&
+                rect.left >= 0 &&
+                rect.bottom <= (window.innerHeight || html.clientHeight) &&
+                rect.right <= (window.innerWidth || html.clientWidth)
+            );
+        }
+
+        //--------------------------------------------------------------------------
+        // Handlers
+        //--------------------------------------------------------------------------
+        /**
+         * Handle clicks outside of the popover to trigger its closing.
+         * Note: if the click event is stopped before bubbling up to the document,
+         * this handler will *not* be fired. Make sure to add custom click handlers
+         * if necessary.
+         * @private
+         * @param {DOMEvent} e
+         */
+        _onDocumentClick(e) {
+            if (!this.popoverRef.el.contains(e.target)) {
```

Comment:
what if the popover class maintains all its instances. hence, at the init of a new one, all the other are destroyed ?

Reply:
that's what was done for `<Dialog>` I believe, so why not. It would prevent having issues with stopped events, since it would be managed through the bus instead